text/0025-core-compute-resources-as-plugins.md

```diff
@@ -0,0 +1,77 @@
+- Feature Name: Core Compute Resources as Plugins
+- Start Date: 2016-10-21
+- RFC PR: (leave this empty)
+- Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Core Compute Resources should move to plugins
+
+# Motivation
+[motivation]: #motivation
+
+Core has been set up to allow compute resources as plugins for some while, and
+we already have several examples of this in use (Xen, Azure, DigitalOcean,
+etc). This has brought benefits:
+
+* able to release independatly of Core
+* make use of the modular Fog gems that are now available
+* separate logging of issues
+* dedicated maintainers who know the compute resource well
+
+In addition, it allows the core project to mark a particular resource as
+unmaintained when necessary (as happened with Xen). 
```

Comment:
I think we agree, but on different parts, as mentioned earlier, non maintained code path should be removed from core (regardless if its compute or not).

regarding maintained code (e.g. vmware/ec2 etc) I think we should move it to a plugin *only* if there is a good reason, I personally don't think that a VMWare maintainer will do better job if its in the repo or in another repo.
This does open the discussion of breaking down maintainer responsibilities, e.g. someone is maintains the vmware code, regardless of where it is, and can make the decision to move it to a plugin if he wishes to.